fmdtools/analyze/graph.py

```diff
@@ -1147,41 +1360,99 @@ def __init__(self, mdl, withstates=True, **kwargs):
         **kwargs : kwargs
             (placeholder for kwargs)
         """
-        self.g=self.nx_from_obj(mdl)
-        if withstates: self.set_nx_states(mdl)
+        self.g = self.nx_from_obj(mdl)
+        if withstates:
+            self.set_nx_states(mdl)
+
     def nx_from_obj(self, mdl):
-        g=mdl.graph.copy()
-        labels = {fname:f.get_typename() for fname, f in mdl.fxns.items()}
-        labels.update({fname:f.get_typename() for fname, f in mdl.flows.items()})
+        """
+        TODO Needs documentation
+        Parameters
+        ----------
+        mdl
+
+        Returns
+        -------
+
+        """
+        g = mdl.graph.copy()
+        labels = {fname: f.get_typename() for fname, f in mdl.fxns.items()}
+        labels.update({fname: f.get_typename() for fname, f in mdl.flows.items()})
         nx.set_node_attributes(g, labels, name='label')
         nx.set_edge_attributes(g, 'contains', name='label')
         return g
+
     def set_nx_states(self, mdl):
+        """
+        TODO : need documentation
+        Parameters
+        ----------
+        mdl
+
+        Returns
+        -------
+
+        """
         self.set_flow_nodestates(mdl)
         self.set_fxn_nodestates(mdl)
+
     def set_fxn_nodestates(self, mdl):
+        """
+        TODO : requires documentation
+        Parameters
+        ----------
+        mdl
+
+        Returns
+        -------
+
+        """
         fxnfaults, fxnstates = {}, {}
         for fxnname, fxn in mdl.fxns.items():
             fxnstates[fxnname] = asdict(mdl.fxns[fxnname].s)
             fxnfaults[fxnname] = [*mdl.fxns[fxnname].m.faults]
         nx.set_node_attributes(self.g, fxnstates, 'states')
         nx.set_node_attributes(self.g, fxnfaults, 'faults')
+
     def set_flow_nodestates(self, mdl):
+        """
+        TODO : requires documentation
+
+        Parameters
+        ----------
+        mdl
+
+        Returns
+        -------
+
+        """
         flowstates = {}
         for flowname, flow in mdl.flows.items():
-            flowstates[flowname]= asdict(flow.s)
+            flowstates[flowname] = asdict(flow.s)
         nx.set_node_attributes(self.g, flowstates, 'states') 
-    def get_multi_edges(self,mdl, subedges):
-        """ Used by subclasses to attach functions/flows (subedges arg) to
-        edges"""
-        flows={}
-        multgraph = nx.projected_graph(mdl.graph, subedges ,multigraph=True)
+
+    def get_multi_edges(self, mdl, subedges):
+        """
+        Used by subclasses to attach functions/flows (subedges arg) to edges
+
+        Parameters
+        ----------
+        mdl
```

Comment:
mdl: Model
    Model to represent.
subedges : list
    nodes from the full graph which will become edges in the subgraph (e.g. individual flows)

Returns
----------
flows : dict
     Dictionary of edges with keys representing each sub-attribute of the edge (e.g., flows)

Reply:
Done